src/mower_msgs/msg/HighLevelStatus.msg

```diff
@@ -12,6 +12,7 @@ uint8 SUBSTATE_SHIFT=6
 uint8 state
 string state_name
 string sub_state_name
+int8 current_area
```

Comment:
maybe 128 mowing areas is a reasonable limit but I would make this 16 or 32 bits .... just in case

Reply:
and/or unsigned, and use 0 when not mowing

Reply:
ok yes I'll change to 16 bit. We never know where this project is going 😆. I had it signed

Reply:
I'm also not totally sure if the submodules update correctly as I have intended so let me know if that bit doesn't work.

Reply:
I've been running it for a bit and it works really nice for implementing start-in-area or mowing specific areas.  In Node Red set up an array of the areas you want to mow in e.g. drop down list, and e.g. [0,3,4] to a variable. Then every time MQTT message for robot_state comes in check if the current_state is MOWING, and if the current_area is not in that list then send skip_area MQTT command.  Works really well and handles pauses in mowing and recharges fine.

Reply:
> ok yes I'll change to 16 bit. We never know where this project is going 😆. I had it signed so that when it is mowing area 0 then it can store 0, and thus be negative for non-mowing situation. Else you have to shift by 1.

ack, got that but also have the state/substate in the msg so don't need negative area to indicate no mowing

Reply:
but 32768 areas should be enough :-)

Reply:
is there a scenario where you might want to save the current mow area, even when not in mowing area, so it doesn't reset to e.g. 0?  If not I'll change from int to uint's.    I have actually changed the implementation as you are probably noticed, to tidy it up and use something similar to the get_state() calls which requires updating all the other behaviour modules.  I think it's building and I'm nearly there now.  If so I'll squash the new commits down and change to uint in a bit

Reply:
np, doesn't really matter now that it's 16bit